index.bs

```diff
@@ -10710,6 +10497,45 @@ property-installation style as namespaces.)
     1.  Return |F|.
 </div>
 
+
+<h5 id="es-default-operations">Default operations</h5>
+
+Only [=regular operations=] which have a <dfn>corresponding default operation</dfn> defined below
+may be declared with a [{{Default}}] [=extended attribute=].
+
+
+<h6 id="es-default-tojson">default toJSON operation</h5>
+
+The [=corresponding default operation=] of the “toJSON” operation is the [=default toJSON operation=].
+
+The [=return type=] of the [=default toJSON operation=] must be {{any}}.
+
+<div algorithm>
+
+    To invoke the <dfn export>default toJSON operation</dfn> of [=interface=] |I|,
+    run the the following steps:
+
+    1.  Let |O| be the <emu-val>this</emu-val> value.
+    1.  Find the closest [=inherited interface=] of |I|
+        that declares a “toJSON” operation
+        or which has a [=consequential interface=] that does.
+        If such an interface exists, then:
+        1.  Let |P| be that interface.
+        1.  If a “toJSON” operation is declared on |P|,
+            then let |super| be that “toJSON” operation.
+        1.  Otherwise, let |super| be the “toJSON” operation
+            declared on one of |P|'s [=consequential interfaces=].
+        1.  Let |result| be [=?=] [=Call=](|super|, |O|).
+        1.  If Type(|result|) is not Object, return |result|.
+    1.  Otherwise, let |result| be [=!=] [=ObjectCreate=]([=%ObjectPrototype%=]).
+    1.  For each [=exposed=] [=regular attribute=] |attr| that is an [=interface member=] of |I|:
+        1.  Let |id| be |attr|’s [=identifier=] [=converted to an ECMAScript value=].
+        1.  Let |value| be [=?=] [=Get=](|O|, |id|).
+        1.  Perform [=?=] [=CreateDataProperty=](|result|, |id|, |value|).
+    1.  Return |result|.
```

Comment:
So I guess we have open questions as to whether this algorithm should observably call the super, or observably perform Get()s. I think it's easy enough to phrase it as doing neither. E.g. to be non-observable instead of

> Let _result_ be ? Call(_super_, _O_)

say something like

> Let _result_ be the result of performing the steps listed in the description of _super_ on _O_ (or the default steps defined for _super_ if _super_ is annotated with [Default])

(and switch around subsequent the type testing to work on IDL values instead of ES values).

And instead of

> Let _value_ be ? Get(_O_, _id_)

say something like

> Let _value_ be the result of performing the steps listed in the description the getter for _id_ on _O_.